compiler/src/iree/compiler/Codegen/LLVMCPU/test/select_x86_64_lowering_strategy.mlir

```diff
@@ -418,11 +418,11 @@ hal.executable private @matmul_partially_pad  {
     }
   }
 }
-//  CHECK-DAG: #[[CONFIG:.+]] = #iree_codegen.lowering_config<tile_sizes = {{\[}}[129, 8, 0], [129, 8, 0], [0, 0, 0], [8, 8, 0], [0, 0, 16], [0, 0, 0]]>
+//  CHECK-DAG: #[[CONFIG:.+]] = #iree_codegen.lowering_config<tile_sizes = {{\[}}[129, 8, 0], [129, 8, 0], [0, 0, 0], [8, 32, 0], [0, 0, 16], [0, 0, 0]]>
```

Comment:
So obviously "clamping" that you are removing impacts cases like this one.

IIUC, this will require masking further down, right? However, this could be avoided by tweaking the distribution sizes - where does "129, 8, 0" come from? I'm mostly curious to learn how likely this scenario is.

Reply:
I think there is a bug in the code. We allow distribution tile sizes to be aligned with power of 2 (or vector size), but the heuristic is wrong. I tried to fix it in https://github.com/openxla/iree/pull/15779, but there are regressions. I don't have cycles to look into them, so the work has been paused. Ideally, they should be `128`. If people accept such regression, then we can move out of this suboptimal state.

https://github.com/openxla/iree/pull/15779